### PR TITLE
adds a way to right tipped over vending machines

### PR DESCRIPTION
**Pull request by Adharainspace**
[CLEANLINESS] [ENHANCEMENT]
## About the PR 
when you hit a vending machine with a crowbar, youll get a 5 second actionbar. if it completes, the vending machine is un-tipped over. i also rearranged the vending file, which apparently git hated. before the structure of the file was like this:
```
- vending machine product datum 
- vending machine vars and like 3 procs
- every vending machine child
- the rest of the vending machine procs
```
i changed it to
```
- vending machine product datum 
- vending machine vars and all the procs
- vending machine action bar datum
- every vending machine child
```

obligatory vid: https://streamable.com/gquhna
also git freaked out at the changes: the lines of code i changed are **424-428** and **938-980** and **2063** 
## Why's this needed? 
its something a lot of new people think should be a thing. i get mhelps about how to right vending machines a lot. and it makes sense. 

## Changelog

```
(u)adharainspace:
(*)you can now fix tipped over vending machines by hitting them with a crowbar!
```

### CODE REVIEW

**Comment by BatElite** [comment]
Excellent

File: code/modules/vending/vending.dm [hunk 1]
```diff
@@ -199,1925 +199,1972 @@
...
+			if(emagged)
+				dat += "<BR><B>Available Credits:</B> CREDIT CALCULATION ERROR<BR>"
+			else
+				dat += "<BR><B>Available Credits:</B> $[src.credit]<BR>"
+			dat += "Topping - <A href='?src=\ref[src];picktopping=1'>[piztopping]</A><BR>"
+			dat += "<A href='?src=\ref[src];cook=1'>Cook!</A><BR>"
+		user.Browse("<HEAD><TITLE>Pizza Vendor</TITLE></HEAD>[dat]", "window=pizzavend")
+		onclose(user, "pizzavend")
+		return
+*/
```
Comment (ZeWaka) [inline code/modules/vending/vending.dm:1650]: delet this

File: code/modules/vending/vending.dm [hunk 2]
```diff
@@ -199,1925 +199,1972 @@
...
-	var/sound_greeting = 'sound/machines/fortune_greeting.ogg'
-	var/sound_greeting_broken = 'sound/machines/fortune_greeting_broken.ogg'
-	var/sound_laugh = 'sound/machines/fortune_laugh.ogg'
-	var/sound_laugh_broken = 'sound/machines/fortune_laugh_broken.ogg'
-	var/sound_ding = 'sound/machines/ding.ogg'
-	var/list/sounds_working = list('sound/misc/automaton_scratch.ogg','sound/machines/mixer.ogg')
-	var/list/sounds_broken = list('sound/machines/glitch1.ogg','sound/machines/glitch2.ogg','sound/machines/glitch3.ogg','sound/machines/glitch4.ogg','sound/machines/glitch5.ogg')
+	onUpdate()
+		..()
+		if(get_dist(src.owner, src.vendor) > 1 || src.vendor == null || src.owner == null)
```
Comment (ZeWaka) [inline code/modules/vending/vending.dm:953]: use DIST_CHECK here

File: code/modules/vending/vending.dm [hunk 3]
```diff
@@ -199,1925 +199,1972 @@
...
 
-	lr = 0.3
-	lg = 0.3
-	lb = 1
-#endif
-	New()
+	onStart()
 		..()
-		light.set_color(0.8, 0.4, 1)
+		if(get_dist(src.owner, src.vendor) > 1 || src.vendor == null || src.owner == null)
```
Comment (ZeWaka) [inline code/modules/vending/vending.dm:963]: DIST_CHECK

**Comment by Adharainspace** [comment]
requesting a review since i implemented your suggestions but github doesnt think i did

**Pull request merged**